### PR TITLE
Fix for half day date not updating.

### PR DESCRIPTION
**Pull request by KirMcC**
endDate value now matches startDate value if:
   - half day is selected
   - endDate is before startDate.

Fixes updating date of half day.

### CODE REVIEW

File: src/components/NewBookingModal/NewBooking/form.js [hunk 1]
```diff
@@ -129,7 +129,17 @@ class RawForm extends Component {
         <DatePicker
           id="startDate"
           selected={values.startDate}
-          onChange={value => setFieldValue('startDate', value)}
+          onChange={value => {
```
Comment (BrownMark) [inline src/components/NewBookingModal/NewBooking/form.js]: Personally I'd separate this out into a separate function. At a glance something like 

dateSelectionChanged = (selectedValue) => {
     if (values.endDate.isBefore(value)) {
              setFieldValue('startDate', value);
              setFieldValue('endDate', value);
            } else if (values.isHalfDay) {
              setFieldValue('endDate', value);
              setFieldValue('startDate', value);
            } else {
              setFieldValue('startDate', value);
            }
}

Althought even better could be to go down the container/component approach.

File: src/components/NewBookingModal/UpdateBooking/form.js [hunk 2]
```diff
@@ -151,7 +151,17 @@ class RawForm extends Component {
         <DatePicker
           id="startDate"
           selected={values.startDate}
-          onChange={value => setFieldValue('startDate', value)}
+          onChange={value => {
```
Comment (BrownMark) [inline src/components/NewBookingModal/UpdateBooking/form.js]: Same as above, but not 100% required.

**Review by BrownMark — APPROVED** [review]
Approved, but there's some debt there.

**Review by d1wilko — APPROVED** [review]
looks good but agree with mark's comments - I appreciate time is of the essence here

File: src/components/NewBookingModal/NewBooking/form.js (hunk 1 above)
Comment (KirMcC) [reply src/components/NewBookingModal/NewBooking/form.js]: Also agree. I can quickly move the function out. 

I do prefer the container/component approach. But `setFieldValue` prop is currently being passed down from `formik`. Not familiar with `formik`, so not 100% sure if I'll be able to extract it to container quickly.

**Comment by KirMcC** [comment]
@d1wilko @BrownMark Function now separated. :)

**Review by d1wilko — APPROVED** [review]
defo better, and feel free to push - but you have an identical function in two files - these could be pulled into a single helper file at some point ;)

File: src/components/NewBookingModal/NewBooking/form.js (hunk 1 above)
Comment (William-Unosquare) [reply src/components/NewBookingModal/NewBooking/form.js]: If it's extracted into its own method you'll have to pass the setFieldValue prop further down as a property when you call it. This means the extracted method would be coupled to forms, which means I don't think it should be extracted personally.

**Review by William-Unosquare — APPROVED** [review]

**Pull request merged**